ndcube/utils/cube.py

```diff
@@ -137,81 +138,98 @@ def get_crop_item_from_points(points, wcs, crop_by_values, keepdims):
+    # pixel coords for each pixel axis. Therefore, to iterate in array axis order,
+    # combined_points_pixel_idx must be reversed.
     item = []
+    ambiguous = []
+    message = ""
     result_is_scalar = True
-    for axis_indices in combined_points_array_idx:
-        if axis_indices == []:
+    for array_axis, pixel_coords in enumerate(combined_points_pixel_idx[::-1]):
+        if pixel_coords == []:
             result_is_scalar = False
             item.append(slice(None))
         else:
-            min_idx = min(axis_indices)
-            max_idx = max(axis_indices) + 1
-            if max_idx - min_idx == 1 and not keepdims:
-                item.append(min_idx)
+            # Calculate the index of the array element containing the pixel coordinate.
+            # Note that integer pixel coordinates correspond to the pixel center,
+            # while integer array indices correspond to lower edge of desired array element.
+            # Therefore a shift of 0.5 is required in the conversion.
+            # The max idx conversion below will discard right-ward array element if
+            # max pixel coord corresponds to a pixel edge.
+            min_array_idx = int(np.floor(min(pixel_coords) + 0.5))
+            max_array_idx = int(np.ceil(max(pixel_coords) - 0.5)) + 1
+            # Due to the above calculation, the above min and max array indices can only be
+            # same if the original pixel coords correspond to the same pixel edge.
+            # If this is the case, increment the max array index by 1 so the rightward array
+            # element is kept. Also, build a warning message about this to be raised later.
+            if min_array_idx == max_array_idx:
+                ambiguous.append(array_axis)
+                message += (f"All input points corresponding to array axis {array_axis} lie on "
+                            f"the boundary between array elements {min_array_idx} and "
+                            f"{max_array_idx}. The cropped NDCube will only include array "
+                            f"element {max_array_idx}.\n")
+                max_array_idx += 1
+            if max_array_idx - min_array_idx == 1 and not keepdims:
+                item.append(min_array_idx)
             else:
-                item.append(slice(min_idx, max_idx))
+                item.append(slice(min_array_idx, max_array_idx))
                 result_is_scalar = False
+    # Raise warning if all world values for any array axes correspond to a pixel edge.
+    if ambiguous:
+        warn_user(message)
     # If item will result in a scalar cube, raise an error as this is not currently supported.
     if result_is_scalar:
         raise ValueError("Input points causes cube to be cropped to a single pixel. "
-                         "This is not supported when keepdims=False.")
+                         "This is not supported when keepdims=False. "
+                         "Please set keepdims=True and try again.")
```

Comment:
I feel that adding this statement is a bit silly.  I feel that it is already implied by the preceding sentence, and it implicitly assumes that the user intended to crop to a single pixel, when it may actually be in error (e.g., specifying only a single world point)